### ISSUE TITLE
listen for changes using inotfiy, stop polling

### ISSUE THREAD

**Issue by michaelficarra** [karl/loris#1]
It looks like you're polling for filesystem changes. That's both gross and a huge waste of resources. I suggest using one of the many [inotify](http://en.wikipedia.org/wiki/Inotify) interfaces for Ruby, such as [rb-inotify](https://github.com/nex3/rb-inotify), [rinotify](http://rubyforge.org/projects/rinotify/), or [ruby-inotify](http://dinhe.net/~aredridel/projects/ruby/ruby-inotify.html).

Worst case scenario, if polling just must be used, `stat -t` lessens resource usage.

**Comment by karl** [comment]
Yeah it is pretty gross, but it was the quickest way to get things working across Windows and OSX while ensuring It picks up newly created files.

Definitely on the look out for a way to make it evented, and be notified of file system changes. But haven't found any gems that let me do everything cross platform (although I haven't looked very hard).

Thanks for your feedback.

**Comment by michaelficarra** [comment]
Mac OS 10.5+ has [FSEvents](http://arstechnica.com/apple/reviews/2007/10/mac-os-x-10-5.ars/7) and Windows has [file/directory change notifications](http://msdn.microsoft.com/en-us/library/aa365261%28VS.85%29.aspx).
